server.ts

```diff
@@ -17,20 +17,77 @@ const dbConfig = {
   connectionString: process.env.DATABASE_URL,
   ssl: sslSetting,
 };
-
+//hello
 const app = express();
 
 app.use(express.json()); //add body parser to each following route handler
 app.use(cors()) //add CORS support to each following route handler
 
+
 const client = new Client(dbConfig);
 client.connect();
 
+// app.use(cors({
+//   origin: '*',
+// methods: ["GET", "POST","PATCH","PUT"]
+// }));
+
+
 app.get("/", async (req, res) => {
-  const dbres = await client.query('select * from categories');
+  res.set('Access-Control-Allow-Origin', '*')
+  const dbres = await client.query('select sub_breed, sum(vote_count) as sumvote_count from vote  group by sub_breed order by sumvote_count desc limit 10');
+
   res.json(dbres.rows);
 });
 
+function urlExtracting(url:{message: string}):string[] {
+ 
+  const breedAndSubBreed= url.message.match(/.*\/(.*)\/(.*)$/)[1]
+  let breed
+  let subBreed
+  if (breedAndSubBreed.includes('-')){
+    breed = breedAndSubBreed.match(/.*(?=-)/)[0]
+    subBreed = breedAndSubBreed.match(/(?<=-).*/)[0]+"-"+breed
+ } else {
+    breed = breedAndSubBreed
+    subBreed = breedAndSubBreed
+ }
+ return [breed,subBreed]
+}
+
+app.post("/", async (req, res) => {
+  // res.set('Access-Control-Allow-Origin', '*')
+  const url = req.body.message
+  const isPresent = await client.query('SELECT CASE WHEN EXISTS (SELECT * FROM vote WHERE image_url = $1)THEN $2 ELSE $3 END',[url,1,0]) //return true if url exists else false
+  
+  
+  if(isPresent.rows[0].case==='0') {
+  const[breed,subBreed]=urlExtracting(url)
+  const dbres = await client.query('insert into vote (breed, sub_breed, image_url,vote_count) values($1,$2,$3,$4) returning sub_breed,id',[breed,subBreed,url,0]);
+  
+  const dbres2 = await client.query('select id from vote where image_url=$1',[url])
```

Comment:
could have given dbres and dbres2 more meaningful names to reflect what they are retrieving from the db